js/app.js

```diff
@@ -126,19 +147,17 @@ const App = {
 			App.$.setEditing($li);
 		});
 		App.todoEvent("keyup", '[data-todo="edit"]', (todo, $li, e) => {
-			let $input = App.$.getTodo($li).$input;
-			if (e.key === "Enter" && $input.value) {
+			if (e.key === "Enter" || e.key === "Escape") {
 				$li.classList.remove("editing");
-				App.todos.update({ ...todo, title: $input.value });
-			}
-			if (e.key === "Escape") {
-				document.activeElement.blur();
 			}
 		});
+		App.todoEvent("input", '[data-todo="edit"]', (todo, $li) => {
+			let $input = App.$.getTodo($li).$input;
+			App.todos.update({ ...todo, title: $input.value });
+		});
+		// Comment out this block to test multiple clones on the same machine
 		App.todoEvent("focusout", '[data-todo="edit"]', (todo, $li) => {
-			if ($li.classList.contains("editing")) {
-				App.render();
-			}
+			$li.classList.remove("editing");
```

Comment:
Here we are changing the behaviour of the app to re-render on every change, not just when an edit is committed by pressing Enter.